python/feldera/sql_context.py

```diff
@@ -0,0 +1,365 @@
+            self.http_input_buffer.append({tbl.name: df.to_dict('records')})
+            return
+
+        tbl = self.todo_tables.get(table_name)
+
+        if not tbl:
+            raise ValueError(f"Table {table_name} not registered")
+
+        # tbl.infer_schema(df)  TODO: support schema inference
+
+        self.tables[table_name] = tbl
+        self.todo_tables.pop(table_name)
+
+        self.http_input_buffer.append({tbl.name: df.to_dict('records')})
+
+    def register_view(self, name: str, query: str):
+        """
+        Registers a Feldera View based on the provided query.
+        Auto inserts the trailing semicolon if not present.
+
+        :param name: The name of the view.
+        :param query: The query to be used to create the view.
+        """
+
+        if query[-1] != ';':
+            query += ';'
+
+        self.views[name] = f"CREATE VIEW {name} AS {query}"
+
+    def listen(self, view_name: str) -> OutputHandler:
+        """
+        Listens to the output of the provided view so that it is available in the notebook / python code.
+
+        :param view_name: The name of the view to listen to.
+        """
+
+        queue = Queue(maxsize=1)
+
+        self.views_tx.append({view_name: queue})
+
+        handler = OutputHandler(self.client, self.pipeline_name, view_name, queue)
+        handler.start()
+
+        return handler
+
+    def from_delta_table(self, table_name: str, connector_name: str, config: dict):
+        """
+        Tells feldera to read the data from the specified delta table.
+
+        :param table_name: The name of the table.
+        :param connector_name: The unique name for this connector.
+        :param config: The configuration for the delta table.
+        """
+
+        # TODO: test this, can't test this right now, because of a serialization issue
+        # TODO: see: https://github.com/feldera/feldera/pull/1764
+
+        if config.get("uri") is None:
+            raise ValueError("uri is required in the config")
+
+        if config.get("mode") is None:
+            raise ValueError("mode is required in the config, valid modes: snapshot, follow, snapshot_and_follow")
+
+        if config.get("mode") not in ["snapshot", "follow", "snapshot_and_follow"]:
+            raise ValueError("mode must be one of snapshot, follow, snapshot_and_follow")
+
+        connector = Connector(name=connector_name,
+                              config={
+                                  "transport": {
+                                      "name": "delta_table_input",
+                                      "config": config,
+                                  }
+                              })
+
+        if table_name in self.input_connectors_buffer:
+            self.output_connectors_buffer[table_name].append(connector)
+        else:
+            self.output_connectors_buffer[table_name] = [connector]
+
+    def to_delta_table(self, view_name: str, connector_name: str, config: dict):
```

Comment:
Hmm, having to pass connector name is annoying. My first instinct is to generate the name internally and not expose it to the user, but that will pollute the namespace with thousands of random connector names. I don't have a better solution yet.